src/base/utils/fs.h

```diff
@@ -33,12 +33,38 @@
  * Utility functions related to file system.
  */
 
+#include <iterator>
+
+#include <QByteArray>
 #include <QString>
 
+class QIODevice;
+
 namespace Utils
 {
     namespace Fs
     {
+        // A wrapper class that satisfy LegacyOutputIterator requirement
+        class FileOutputIterator
+            : public std::iterator<std::output_iterator_tag, void, void, void, void>
+        {
+        public:
+            explicit FileOutputIterator(QIODevice &device, const int bufferSize = (4 * 1024));
+            ~FileOutputIterator();
+
+            // mimic std::ostream_iterator behavior
+            FileOutputIterator &operator=(const char c);
+            // TODO: make these `constexpr` in C++17
```

Comment:
For the record, https://stackoverflow.com/questions/36489369/constexpr-non-static-member-function-with-non-constexpr-constructor-gcc-clang-d#comment96878182_36489493